pysoem/pysoem.pyx

```diff
@@ -934,6 +944,29 @@ cdef class CdefSlave:
                    wd_time_reg.to_bytes(2, byteorder='little', signed=False),
                    fprd_fpwr_timeout_us)
 
+    def add_emergency_callback(self, callback):
+        """Get notified on EMCY messages from this slave.
+
+        :param callback:
+            Callable which must take one argument of an
+            :class:`~Emergency` instance.
+        """
+        self._emcy_callbacks.append(callback)
+
+    cdef _on_emergency(self, cpysoem.ec_errort* emcy):
+        """Notify all emergency callbacks that an emergency message
+        was received.
+
+        :param emcy: Emergency object.
+        """
+        for callback in self._emcy_callbacks:
+            callback(Emergency(emcy.Slave, 
```

Comment:
@MartinALL91 I also wonder why someone would need multiple emergency message consumers? You probably would collect them all in one central place.

Reply:
I think it's more flexible to be able to subscribe with multiple callbacks. It is how the CANopen [EmcyConsumer](https://github.com/christiansandberg/canopen/blob/master/canopen/emcy.py#L36) works.
If you prefer to have a single callback, I could make that change.